Migration of the RETORCH Annotations to Github close #6

augustocristian opened the pull request:
- Created classes with the custom annotations
- Improved grammar and spell of the original comments

augustocristian commented:
Its necesary to remove Class2 and its test of the repository prior to merge this pull request? @javiertuya

javiertuya commented on retorch-annotations/src/main/java/giis/retorch/annotations/AccessMode.java:
```diff
@@ -0,0 +1,29 @@
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+//Copyright 2020, Cristian Augusto, All rights reserved.
```
Remove, we haven't a header yet, and copyright is uniovi

javiertuya commented on retorch-annotations/src/main/java/giis/retorch/annotations/AccessMode.java:
```diff
@@ -0,0 +1,29 @@
+//Copyright 2020, Cristian Augusto, All rights reserved.
+
+/**
+ * The RETORCH custom annotation for Access Mode has the following attributes:
```
This comment does not provides additional useful information, see https://github.com/giis-uniovi/.github/blob/main/profile/CONTRIBUTING.md.
Comment should describe what is an access mode

javiertuya commented on retorch-annotations/src/main/java/giis/retorch/annotations/AccessMode.java:
```diff
@@ -0,0 +1,29 @@
+
+/**
+ * The RETORCH custom annotation for Access Mode has the following attributes:
+ * resID: A String representing the resource identifier, with a default value of "None."
+ * sharing: A Boolean indicating if sharing is enabled, with a default value of false.
+ * concurrency: An Integer specifying the concurrency level, with a default value of -1.
+ * accessMode: A String representing the access mode, with a default value of "NOASSIGNED"
```
Similar comment than previous line. In this case, a good informative comment saying what is each attribute would be useful to the developer if the comments are placed at each attribute (to see their meaning when they write)

javiertuya commented on retorch-annotations/src/main/java/giis/retorch/annotations/AccessMode.java:
```diff
@@ -0,0 +1,29 @@
+    int concurrency() default -1;
+
+    String accessMode() default "NOASSIGNED";
+
+
```
Avoid unnecesary spacing

javiertuya reviewed (changes requested):
There are no tests, that make unused many things in the project  In addition to the comments placed in the annotations package:

Review all files in the project (both parent and module) to remove everything that is not needed (poms, workflows, remaining classes...).

javiertuya commented:
@augustocristian Please, do not mention for every comment that is resolved. Simply mark it as resolved. When everything is resolved, request review. At this moment I will receive a TODO to be aware that I must start reviewing

augustocristian commented:
> @augustocristian Please, do not mention for every comment that is resolved. Simply mark it as resolved. When everything is resolved, request review. At this moment I will receive a TODO to be aware that I must start reviewing

I apologize for any confusion. I didn't expect the commits to also notify the user if they are mentioned in them. Regarding the workflows, I have one question:
If I remove or deactivate workflows that are not used (e.g., test-java), the pull request doesn't pass because it's required. Am I misunderstanding what to do?
The idea is to have two minimal pom.xml files, each containing only the Maven stages that are useful. For example, the retorchanotations pom.xml should not have any dependencies related to testing, such as jacoco or logging.

javiertuya commented:
> If I remove or deactivate workflows that are not used (e.g., test-java), the pull request doesn't pass because it's required. Am I misunderstanding what to do?
> The idea is to have two minimal pom.xml files, each containing only the Maven stages that are useful. For example, the retorchanotations pom.xml should not have any dependencies related to testing, such as jacoco or logging.

@augustocristian The test workflow should be kept, but only to compile (you may rename to build.yml if you prefer, but when other modules be included you would need to rename again). So, there is no problem, the test workflow triggers sonarqube, and everything is as before.

Regarding poms, exactly, testing/reporting and logging dependencies should be removed

augustocristian commented:
I have noticed another requested change that is marked as unresolved when all is solved-marked :( . I have modified the workflows to remove the required artifacts and changed the Maven command to compile.

javiertuya commented:
@augustocristian FYI, Artifact published in maven central (version 1.1.0)
